packages/amplify_authenticator/lib/amplify_authenticator.dart

```diff
@@ -202,28 +207,54 @@ class Authenticator extends StatefulWidget {
 
 class _AuthenticatorState extends State<Authenticator> {
   final AuthService _authService = AmplifyAuthService();
-
   late final StateMachineBloc _stateMachineBloc;
+  AmplifyConfig? _config;
+  late List<String> _missingConfigValues;
+  bool _configInitialized = false;
 
   @override
   void initState() {
     super.initState();
     _stateMachineBloc = StateMachineBloc(_authService)
       ..authEvent.add(GetCurrentUser());
+    waitForConfiguration();
+  }
+
+  Future<void> waitForConfiguration() async {
+    await Amplify.asyncConfig.then((config) {
+      setState(() {
+        _config = config;
+        _configInitialized = true;
+        _missingConfigValues = missingConfigValues(config);
+      });
+    });
+  }
+
+  List<String> missingConfigValues(AmplifyConfig? config) {
+    List<String> missingValues = [];
+    var cognitoPlugin = config?.auth?.awsCognitoAuthPlugin?.auth?['Default'];
+    cognitoPlugin ?? missingValues.add('auth.plugins.Auth.Default');
+    cognitoPlugin?.signupAttributes?.length ??
+        missingValues.add('auth.plugins.Auth.Default.signUpAttributes');
+    return missingValues;
   }
 
   @override
   Widget build(BuildContext context) {
     var defaultForms = DefaultForms();
     AuthStringResolver resolver = widget.resolver ?? AuthStringResolver();
-
     defaultForms.context = context;
```

Comment:
This is generally an anti-pattern. At some point, we'll want to rewrite this logic.

Reply:
Yeah - I think I am just going to bite the bullet and widget-ify the default forms, which will obviate the need for this.